### PR TITLE
Feature/tip management

### PR DESCRIPTION
**Pull request by vc-vishva**
tip - management task using typescript

### CODE REVIEW

File: tip-management-TS/src/models/tips.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+const placeSchema: Schema<IPlace> = new Schema(
+  {
+    placeName: { type: String, required: true },
+    billAmount: { type: Number, required: true },
+    tipAmount: { type: Number, required: true },
+    createdAt: { type: Date, default: Date.now },
+    updatedAt: { type: Date, default: Date.now },
+    user_id: { type: String, required: true },
```
Comment (vc-mohit) [inline tip-management-TS/src/models/tips.ts:19]: take ref from User Schema, this is not correct.

File: tip-management-TS/src/types/type.ts
Comment (vc-mohit) [inline tip-management-TS/src/types/type.ts:1]: this interfaces is required?

File: tip-management-TS/src/services/userService.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const saltRounds = 10;
+  return await bcrypt.hash(password, saltRounds);
+};
+
+export const createUser = async (userData: {
+  f_name: string;
+  l_name: string;
+  email: string;
+  password: string;
+}): Promise<IUser> => {
```
Comment (vc-mohit) [inline tip-management-TS/src/services/userService.ts:20]: why you destructure userdata object in params?

File: tip-management-TS/src/controller/userController.ts
Comment (vc-mohit) [inline tip-management-TS/src/controller/userController.ts:1]: need to change lot's of thing, we will discuss on this.

**Review by vc-mohit — CHANGES_REQUESTED** [review]
remove all consoles,
remove unused imports
remove commented code

File: tip-management-TS/src/services/userService.ts (hunk 2 above)
Comment (vc-vishva) [reply tip-management-TS/src/services/userService.ts:20]: extract specific properties from the userdata object directly in the function parameter.

File: tip-management-TS/src/controller/userController.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    if (!passwordMatch) {
+      return apiResponse(res, 401, "Invalid email or password");
+    }
+
+    const token = jwt.sign({ id: user._id }, process.env.JWT_SECRET as Secret, {
+      expiresIn: "24h",
+    });
+    apiResponse(res, 201, "Success", [token]);
+  } catch (error) {
+    console.log("Error:", error);
```
Comment (vc-mohit) [inline tip-management-TS/src/controller/userController.ts:60]: remove console

File: tip-management-TS/src/middleware/verify.ts
Comment (vc-mohit) [inline tip-management-TS/src/middleware/verify.ts:1]: remove all consoles

File: tip-management-TS/src/utils/response.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { Response } from "express";
+
+export const replaceMsg = (message: string, data: string) => {
```
Comment (vc-mohit) [inline tip-management-TS/src/utils/response.ts:3]: what is the use of this function?

File: tip-management-TS/src/validation/tipValidation.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { body } from "express-validator";
+
+export const validatePlace = [
+  body("placeName")
+    .notEmpty()
+    .withMessage("Place name is required")
+    .isLength({ min: 25 })
```
Comment (vc-mohit) [inline tip-management-TS/src/validation/tipValidation.ts:7]: is the code working?

File: tip-management-TS/src/validation/userValidation.ts
Comment (vc-mohit) [inline tip-management-TS/src/validation/userValidation.ts:1]: same for this file

File: tip-management-TS/src/utils/apiError.ts
Comment (vc-mohit) [inline tip-management-TS/src/utils/apiError.ts:1]: this file is required?

File: tip-management-TS/src/validation/userValidation.ts
Comment (vc-vishva) [reply tip-management-TS/src/validation/userValidation.ts:1]: yes

File: tip-management-TS/src/validation/tipValidation.ts
Comment (vc-mohit) [inline tip-management-TS/src/validation/tipValidation.ts:1]: validation is not proper?

File: tip-management-TS/src/middleware/verify.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      const user = await User.findById(decoded.id);
+      if (!user) {
+        return apiResponse(res, 401, "not found");
+      }
+
+      req.id = decoded.id;
+      next();
+    }
+  } catch (error) {
+    res.status(401).send({ error: error });
```
Comment (vc-mohit) [inline tip-management-TS/src/middleware/verify.ts:35]: why you are sending 2 times res?

File: tip-management-TS/src/middleware/verify.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import User from "../models/users";
+import { Express, Request, Response, NextFunction } from "express";
+import jwt, { Secret, JwtPayload } from "jsonwebtoken";
+import apiResponse from "../utils/response";
+import { ObjectId } from "mongoose";
+
+export interface test {
```
Comment (vc-mohit) [inline tip-management-TS/src/middleware/verify.ts:7]: change the name of Interface.

**Review by vc-mohit — CHANGES_REQUESTED** [review]
do all the changes, then I'll merge the PR.

**Review by vc-mohit — APPROVED** [review]

**Pull request merged**